sklearn/multiclass.py

```diff
@@ -1069,3 +1240,28 @@ def predict(self, X):
         ).T
         pred = pairwise_distances_argmin(Y, self.code_book_, metric="euclidean")
         return self.classes_[pred]
+
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.4
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+
+        router = (
+            MetadataRouter(owner=self.__class__.__name__)
+            .add_self_request(self)
```

Comment:
So, if any method from the class takes anything else than `(self, X, y, **{method}_params)`, then we will do `.add_self_request(self)`? 

For instance, in the case of the multimetric classifiers it would be `partial_fit` consuming `classes`, that would make `.add_self_request(self)` necessary, correct?